tests/contrib/langchain/test_langchain_community.py

```diff
@@ -1100,6 +1101,114 @@ def test_embedding_logs_when_response_not_completed(
+    prompt = langchain_core.prompts.ChatPromptTemplate.from_messages(
+        [("system", "You are world class technical documentation writer."), ("user", "{input}")]
+    )
+    llm = langchain_openai.OpenAI()
+
+    chain = prompt | llm
+    with request_vcr.use_cassette("lcel_openai_chain_call.yaml"):
+        chain.invoke({"input": "how can langsmith help with testing?"})
+
+
+@pytest.mark.snapshot
+def test_lcel_chain_complicated(langchain_core, langchain_openai, request_vcr):
+    prompt = langchain_core.prompts.ChatPromptTemplate.from_template(
+        "Tell me a short joke about {topic} in the style of {style}"
+    )
+
+    chat_openai = langchain_openai.ChatOpenAI()
+    openai = langchain_openai.OpenAI()
+
+    model = chat_openai.configurable_alternatives(
+        langchain_core.runnables.ConfigurableField(id="model"),
+        default_key="chat_openai",
+        openai=openai,
+    )
+
+    chain = (
+        {
+            "topic": langchain_core.runnables.RunnablePassthrough(),
+            "style": langchain_core.runnables.RunnablePassthrough(),
+        }
+        | prompt
+        | model
+        | langchain_core.output_parsers.StrOutputParser()
+    )
+
+    with request_vcr.use_cassette("lcel_openai_chain_call_complicated.yaml"):
+        chain.invoke({"topic": "chickens", "style": "a 90s rapper"})
+
+
+@pytest.mark.asyncio
+@pytest.mark.snapshot
+async def test_lcel_chain_simple_async(langchain_core, langchain_openai, request_vcr):
+    prompt = langchain_core.prompts.ChatPromptTemplate.from_messages(
+        [("system", "You are world class technical documentation writer."), ("user", "{input}")]
+    )
+    llm = langchain_openai.OpenAI()
+
+    chain = prompt | llm
+    with request_vcr.use_cassette("lcel_openai_chain_acall.yaml"):
+        await chain.ainvoke({"input": "how can langsmith help with testing?"})
+
+
+@pytest.mark.snapshot
+def test_lcel_chain_batch(langchain_core, langchain_openai, request_vcr):
+    """
+    Test that invoking a chain with a batch of inputs will result in a 4-span trace,
+    with a root RunnableSequence span, then tree LangChain ChatOpenAI spans underneath
+    """
+    prompt = langchain_core.prompts.ChatPromptTemplate.from_template("Tell me a short joke about {topic}")
+    output_parser = langchain_core.output_parsers.StrOutputParser()
+    model = langchain_openai.ChatOpenAI()
+    chain = {"topic": langchain_core.runnables.RunnablePassthrough()} | prompt | model | output_parser
+
+    with request_vcr.use_cassette("lcel_openai_chain_batch.yaml"):
+        chain.batch(["chickens", "cows", "pigs"])
+
+
+@pytest.mark.snapshot
+def test_lcel_chain_nested(langchain_core, langchain_openai, request_vcr):
+    """
+    Test that invoking a nested chain will result in a 4-span trace with a root
+    RunnableSequence span, then another RunnubaleSequence and LangChain ChatOpenAI call,
+    and finally a second LangChain ChatOpenAI call.
+    """
```

Comment:
I think I'm slightly confused here. Shouldn't this result in a 5-span trace with:
1. Overall `complete_chain` span
2. Chain 1
3. Chain 2
4. OpenAI call from chain 1
5. OpenAI call from chain 2?

Reply:
yea you're right that my description is no good - not sure why I wrote it like that (taking a look at the snapshot file, it does seem correct, but I don't think the snapshot file that was generated includes the OpenAI calls). Based on the image in the PR description, and just kinda tracing it out by hand, we would end up with something like

1. Overall `complete_chain` span (`RunnableSequence` one)
2. `chain1` (`RunnableSequence` two)
3. `langchain_openai` call from `chain1`
4. `openai` call from `chain1`
5. `langchain_openai` call from `complete_chain` (since `chain2` is actually not evaluated here as a whole chain, but instead its components are evaluated as part of `complete_chain`, it's never even called in langchain internals as it's own `RunnableSequence`)
6. `openai` call from `complete_chain`

like I said though, I think something is wrong with the snapshot generated (as the trace itself that uses the same code looks fine). let me see what's going on here